pkg/controller/deployment/deployment_controller.go

```diff
@@ -491,6 +507,39 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return dc.syncStatusOnly(d)
 	}
 
+	// Handle overlapping deployments by deterministically avoid syncing deployments that fight over ReplicaSets.
```

Comment:
Can you extract all this diff into a `dc.isOverlapping` method in order to keep `syncDeployment` clean?